### DISCUSSION TITLE
Hover panel for standard search results

### DISCUSSION THREAD

**Discussion by JoeAyre** [microsoft-search/pnp-modern-search#2305] [Ideas]
Does anyone have any advice on whether it would be possible to introduce a hover panel for standard search results? I know there is hover functionality in the 'People' layout, but wondered if anything similar existed for non-people search results.

What I am trying to achieve is as follows: I'm using the 'Detailed list' layout to show search results. I'm comfortable enough to get custom columns displaying, however as soon as you add four or more columns, the real estate of the page (i.e. the width) quickly gets filled. I would like to introduce a hover panel so that when a user hovers on a particular search result, more information is presented to them.

I'm sure this is something of an edge case requirement, however I do think it would be a really useful addition to the pnp modern search solution as it may become something that is used in other areas of a SharePoint site, almost like a super customisable highlighted content web part.

**Comment by wobba** [comment]
Developing a custom web component is one approach and use that in your template. https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_web_component/

**Comment by JoeAyre** [comment]
Thanks @wobba - appreciate your response and suggestion. It looks like there are plenty of ways to extend the solution if I had a developer skillset. Whilst I can just about follow examples and tweak them to get roughly where I want to get to, I'm not an out and out developer so attempting to create a hover panel is going to be beyond me. I wonder if you know of any examples/walkthroughs that demonstrate an example of implementing a custom web component that you could point me to?

**Reply by wobba** [reply]
Without a developer you cannot get onmouse/onclick events to trigger separate UX. JavaScript events are explicitly blocked in templates as scripts are a security issue, thus any script logic has to be achieved via extension code which can be properly controlled and deployed.
